cmd/dep/status.go

```diff
@@ -639,7 +643,49 @@ func formatVersion(v gps.Version) string {
 	return v.String()
 }
 
-func collectConstraints(ptree pkgtree.PackageTree, p *dep.Project, sm gps.SourceManager) map[string][]gps.Constraint {
-	// TODO
-	return map[string][]gps.Constraint{}
+// projectConstraint stores ProjectRoot and Constraint for that project.
+type projectConstraint struct {
+	Project    gps.ProjectRoot
+	Constraint gps.Constraint
+}
+
+// constraintsCollection is a map of ProjectRoot(dependency) and a collection of
+// projectConstraint for the dependencies. This can be used to find constraints
+// on a dependency and the projects that apply those constraints.
+type constraintsCollection map[string][]projectConstraint
+
+// collectConstraints collects constraints declared by all the dependencies.
+func collectConstraints(ctx *dep.Ctx, p *dep.Project, sm gps.SourceManager) constraintsCollection {
+	constraintCollection := make(constraintsCollection)
+
+	// Get direct deps of the root project.
+	_, directDeps, err := getDirectDependencies(sm, p)
+	if err != nil {
+		ctx.Err.Println("Error getting direct deps:", err)
+	}
+	// Create a root analyzer.
+	rootAnalyzer := newRootAnalyzer(false, ctx, directDeps, sm)
+
+	// Iterate through the locked projects and collect constraints of all the projects.
+	for _, proj := range p.Lock.Projects() {
+		manifest, _, err := sm.GetManifestAndLock(proj.Ident(), proj.Version(), rootAnalyzer)
+		if err != nil {
+			ctx.Err.Println("Error getting manifest and lock:", err)
+			continue
+		}
+
+		// Get project constraints.
+		pc := manifest.DependencyConstraints()
+
+		// Iterate through the project constraints to get individual dependency
+		// project and constraint values.
+		for pr, pp := range pc {
+			constraintCollection[string(pr)] = append(
```

Comment:
so this is overzealous right now - it pulls in constraints regardless of whether they're actually applicable or not. it's unlikely to create problems right now, but it will need to be updated in the future (please open an issue!) to do the equivalent of something like [`intersectConstraintsWithImports()`](https://github.com/golang/dep/blob/7ccbfed88623a8565f300d02fe76c0c3a1daaac8/gps/solver.go#L740) or [`getApplicableConstraints()`](https://github.com/golang/dep/blob/7ccbfed88623a8565f300d02fe76c0c3a1daaac8/gps/rootdata.go#L82)